### ISSUE TITLE
Using word2vec

### ISSUE THREAD

**Issue by YoadTew** [pytorch/fairseq#3404]
Hey,

I'm trying to create a loss function based on embedding distance from the target word for translation (to replace CE). 
Iv'e used the 'examples/translation' code to create my dataset and vocabulary based on the "prepare-iwslt14.sh".
It created in data-bin the dictionary files and and .bin and .idx for the sentences.

How can I run word2vec over the target vocabulary, using some corpus, to create the word2vec embeddings?
I want the token split to be the same as the one created by fairseq-preprocess, and the token indexes to be consistent between word2vec and fairseq.

Thanks!

**Comment by GayaniNan** [comment]
Hello,

I'm also facing a similar issue with word2vec embeddings. Were you able to resolve it?

Thanks in advance.
